x/ibc-rate-limit/contracts/rate-limiter/src/state.rs

```diff
@@ -0,0 +1,149 @@
+use cosmwasm_std::{Addr, Timestamp};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::cmp;
+
+use cw_storage_plus::{Item, Map};
+
+use crate::msg::QuotaMsg;
+
+pub const RESET_TIME_DAILY: u64 = 60 * 60 * 24;
+pub const RESET_TIME_WEEKLY: u64 = 60 * 60 * 24 * 7;
+pub const RESET_TIME_MONTHLY: u64 = 60 * 60 * 24 * 30;
+
+#[derive(Debug, Clone)]
+pub enum FlowType {
+    In,
+    Out,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema, Copy)]
+pub struct Flow {
+    pub inflow: u128,
+    pub outflow: u128,
+    pub period_end: Timestamp,
+}
+
+impl Flow {
```

Comment:
Can we document what this is exactly please? I'm not very familiar with IBC concepts so it would be amazing having docs for some of these abstractions

Reply:
yes! a flow isn't really an IBC concept but something specific to rate limiting, but I'll add comments (and maybe a readme?) to document this.

Reply:
Added some docs to `state.rs`